km63/Onishchenko_Andrii/laboratory1.sql

```diff
@@ -1,2 +1,137 @@
--- LABORATORY WORK 1
--- BY Onishchenko_Andrii
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+робити вибірки з таблиць та оновлення даних у таблицях
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER onishchenko IDENTIFY BY onishchenko;
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP"
+ALTER USER onishchenko QUOTE 100M ON USERS;
+GRANT "CONNECT" TO onishchenko
+GRANT SELECT ANY TABLE TO onsihchenko
+GRANT UPDATE ANY TABLE TO onsihchenko
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент вивчає мови програмування.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE TABLE table1 ( table1_name VARCHAR2(30) NOT NULL);
+ALTER TABLE table1
+ADD CONSTRAINT table1_name_pk PRIMARY KEY (table1_name);
+
+CREATE TABLE table2 ( table2_name VARCHAR2(30) NOT NULL);
+ALTER TABLE table2
+ADD CONSTRAINT table2_name_pk PRIMARY KEY (table2_name);
+
+CREATE TABLE tablex (
+tablex1_name VARCHAR(30) NOT NULL
+tablex2_name VARCHAR(30) NOT NULL
+tablenumbers NUMBER(8,3) NOT NULL
+);
+
+ALTER TABLE tablex 
+ADD CONSTRAINT table_pk PRIMARY KEY  (table1_name_pk, table2_name_pk)
```

Comment:
`PRIMARY KEY  (table1_name_pk, table2_name_pk)`
Wrong fields attached to the key